common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -3486,6 +3486,13 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         " it is empty, which means that all the connections to HiveServer2 are authenticated. " +
         "When it is non-empty, the client has to provide a Hive user name. Any password, if " +
         "provided, will not be used when authentication is skipped."),
+    HIVE_SERVER2_TRUSTED_DOMAIN_USE_XFF_HEADER("hive.server2.trusted.domain.use.xff.header", false,
```

Comment:
Instead of adding a new config, can we always get the client IP address from the forwarded addresses when they are present in the header? Do you see any use of this config be when it's false?

Reply:
adding it by default will let clients spoof with XFF headers and some proxies might now sanitize it correctly. In most cases, proxies will use the client's ip to connect to HS2 (reverse proxies) in which case we can just use request.getRemoteHost() and not rely on XFF.